apps/sms/views/shared/test/unit/contacts_test.js

```diff
@@ -30,119 +30,7 @@ suite('Contacts', function(done) {
-          }, {})
-        });
-
-        // default value
-        var result = MockContact.list();
-
-        // Supports the error case
-        if (filter == null ||
-            (!filter.filterOp || !filter.filterValue)) {
-          result = null;
-        }
-
-        if (filter.filterValue === targetLocalNumber ||
-            filter.filterValue === targetLocalEmail) {
-          result = [{
-            name: [localContactName]
-          }];
-        }
-
-        if (filter.filterValue === targetFbNumber ||
-            filter.filterValue === notFoundNumber ||
-              filter.filterValue === notFoundEmail) {
-          result = [];
-        }
-
-        // Supports two "no match" cases
-        if (filter.filterValue === '911' ||
-            filter.filterValue === 'wontmatch' ||
-              filter.filterValue === 'z@y.com') {
-          result = [];
-        }
-
-        if (filter.filterValue === 'jane') {
-          result = MockContact.list([
-            // true
-            { givenName: ['Jane'], familyName: ['D'] },
-            // false
-            { givenName: ['jane'], familyName: ['austen'] },
-            // true
-            { givenName: ['jane'], familyName: ['doe'] },
-            // false
-            { givenName: ['jane'], familyName: ['fonda'] },
-            // true
-            { givenName: ['jane'], familyName: ['dow'] },
-            // false
-            { givenName: ['janet'], familyName: [''] }
-          ]);
-        }
-
-        if (filter.filterValue === 'julien') {
-          result = MockContact.list([
-            { givenName: ['Julien'] }
-          ]);
-        }
-
-        if (filter.filterValue === 'do') {
-          result = MockContact.list([
-            // true
-            { givenName: ['Jane'], familyName: ['Doozer'] },
-            // false
-            { givenName: ['doug'], familyName: ['dooley'] },
-            // true
-            { givenName: ['jane'], familyName: ['doe'] },
-            // false
-            { givenName: ['jerry'], familyName: ['doe'] },
-            // true
-            { givenName: ['j'], familyName: ['dow'] },
-            // true
-            { givenName: ['john'], familyName: ['doland'] }
-          ]);
-        }
-
-        if (filter.filterValue === 'doozer') {
-          result = MockContact.list([
-            { givenName: ['Jane'], familyName: ['Doozer'] }
-          ]);
-        }
-
-        if (filter.filterValue === 'mary') {
-          result = MockContact.list([
-            { givenName: ['Mary Anne'], familyName: ['Jones'] }
-          ]);
-        }
-
-        if (filter.filterValue === 'callonerror') {
-          result = null;
-        }
-
-        // All other cases
-
-        if (result === null) {
-          return Promise.reject(
-            new Error('Mock missing filter params')
-          );
-        }
-
-        return Promise.resolve(result);
-      }
-    };
+    sinon.stub(navigator.mozContacts, 'find');
```

Comment:
please use `this.sinon.stub`; `this.sinon` is a sinon sandbox that is automatically restored at the end of a test, so you don't need to restore manually.